python_modules/dagster/dagster/_cli/asset.py

```diff
@@ -78,7 +98,27 @@ def execute_materialize_command(instance: DagsterInstance, kwargs: Mapping[str,
                 "All selected assets must have a PartitionsDefinition containing the passed"
                 f" partition key `{partition}` or have no PartitionsDefinition."
             )
-
+    elif partition_range_start:
```

Comment:
Would be good to add a check/exception here to see if the current backfill policy is single run, & error if not by checking the `asset.backfill_policy.policy_type`